core/src/main/java/org/apache/druid/data/input/FirehoseFactory.java

```diff
@@ -77,4 +77,14 @@ default boolean isSplittable()
   {
     return false;
   }
+
+  /**
+   * Some FirehoseFactory implementations require additional context from the task that uses them.
+   * For example, IngestSegmentFirehoseFactory needs a TaskToolbox from the indexing-service task.
+   * This method is also implemented by implementations such as CombiningFirehoseFactory which wrap
+   * other implementations.
+   */
+  default void setContext(String key, Object value)
```

Comment:
Do you have any plan to add other stuffs in addition to `taskToolBox`? If not, it looks too general to me.

Also, what do you think about adding a new parameter for `taskToolBox` to `connect()`? Even though It's currently used only in `IngestSegmentFirehoseFactory.connect()`, the delegating firehoseFactories should pass them, so I think it makes sense.

Reply:
I'm fine to just make it setTaskToolbox. But I had trouble making the argument be declared TaskToolbox rather than Object due to which module TaskToolbox is declared in.

Re adding it to `connect()`, that won't help for the use case where we want a TaskToolbox to determine splits (which is my actual motivation here)...